src/main/java/com/google/sps/servlets/ElectionServlet.java

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+@WebServlet("/election")
+public class ElectionServlet extends HttpServlet {
+
+  private static final String baseURL = "https://www.googleapis.com/civicinfo/v2/elections?key=";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        StringBuilder strBuf = new StringBuilder();  
+        
+        HttpURLConnection conn = null;
+        BufferedReader reader = null;
+
+        try {  
+            URL url = new URL(baseURL + "AIzaSyBlNN_uUrbqY2uRaBpKTD-5Jbw6nk8n0_k");
```

Comment:
Is there any reason why we need to keep the key portion separate from the baseURL?

Reply:
I think it will work with them as separate after we move the raw key to a separate config file, and then bring it in as a constant (afaik API keys should be hidden from the public files)

Reply:
^ my intent in keeping them separate was to resolve retrieving the API from a separate source, as Casey said.